python/lsst/cp/pipe/defects.py

```diff
@@ -539,48 +546,68 @@ def findHotAndColdPixels(self, exp, imageType, setMask=False):
+                continue
             x.append(bbox.getMinX())
             y.append(bbox.getMinY())
         x = np.array(x)
         y = np.array(y)
-
+        stop
         # Find the defects with same "x" (vertical) coordinate (column).
         unique, counts = np.unique(x, return_counts=True)
-        colThreshold = self.config.colThreshold
-        gapThreshold = self.config.gapThreshold  # (not "bad" pixels, see below).
+        badPixelColumnThreshold = self.config.badPixelColumnThreshold
         multipleX = []
         for (a, b) in zip(unique, counts):
-            if b > colThreshold:
+            if b >= badPixelColumnThreshold:
                 multipleX.append(a)
+        print ("multipleX: ", multipleX)
         if len(multipleX) != 0:
-            for x0 in multipleX:
-                index = np.where(x == x0)
-                multipleY = y[index]  # multipleY and multipleX are in 1-1 correspondence.
-                minY, maxY = np.min(multipleY), np.max(multipleY)
-
-                # Next few lines: don't mask pixels in column if gap between
-                # two consecutive bad pixels is too large.
-                diffIndex = np.where(np.diff(multipleY) > gapThreshold)[0]
-                if len(diffIndex) != 0:
-                    limits = [minY]  # put the minimum first
-                    for gapIndex in diffIndex:
-                        limits.append(multipleY[gapIndex])
-                        limits.append(multipleY[gapIndex+1])
-                    limits.append(maxY)  # maximum last
-                    assert (len(limits)%2 == 0)  # `limits` it's even by design, but check anyways
-                    for i in np.arange(0, len(limits)-1, 2):
-                        s = Box2I(minimum = Point2I(x0, limits[i]), maximum = Point2I(x0, limits[i+1]))
-                        defects.append(s)
-                else:  # No gaps large enough
-                    s = Box2I(minimum = Point2I(x0, minY), maximum = Point2I(x0, maxY))
-                    defects.append(s)
+            defects = self._markOneBadColumn (x, y, x0, multipleX, multipleY, defects)
+        
+        return defects
 
+    def _markOneBadColumn (x, y, x0, multipleX, multipleY):
+        """
+        Add documentation here Placitas
+        """
+        goodPixelColumnGapThreshold = self.config.goodPixelColumnGapThreshold  # (not "bad" pixels, see below).
+        for x0 in multipleX:
+            index = np.where(x == x0)
+            multipleY = y[index]  # multipleY and multipleX are in 1-1 correspondence.
+            minY, maxY = np.min(multipleY), np.max(multipleY)
+
+            # Next few lines: don't mask pixels in column if gap of good pixels between
+            # two consecutive bad pixels is larger or equal than 'goodPixelColumnGapThreshold'.
+            diffIndex = np.where(np.diff(multipleY) >= goodPixelColumnGapThreshold)[0]
+            print ("diffIndex: ", diffIndex)
```

Comment:
We don't use `print()` functions in production code - replace with logger messages.